tests/integration/test_scaling.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+import time
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+from connection_check import broker_active_check, brokers_active_get
+from tests.integration.helpers import get_kafka_zk_relation_data
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = "kafka"
+ZK = "zookeeper"
+DUMMY_NAME_1 = "app"
+
+
+@pytest.mark.abort_on_fail
+async def test_kafka_simple_scale_up(ops_test: OpsTest):
+    kafka_charm = await ops_test.build_charm(".")
+
+    await asyncio.gather(
+        ops_test.model.deploy(
+            "zookeeper", channel="edge", application_name="zookeeper", num_units=1
+        ),
+        ops_test.model.deploy(kafka_charm, application_name=APP_NAME, num_units=1),
+    )
+    await ops_test.model.wait_for_idle(apps=[APP_NAME, ZK])
+    await ops_test.model.add_relation(APP_NAME, ZK)
+    await ops_test.model.wait_for_idle(apps=[APP_NAME, ZK])
+    assert ops_test.model.applications[ZK].status == "active"
+    assert ops_test.model.applications[APP_NAME].status == "active"
+
+    await ops_test.model.applications[APP_NAME].add_units(count=3)
```

Comment:
It might be better to add only 2 units here. Maybe even only 1. 

GitHub CI is really slow, and having lots of active units causes some problems.

Reply:
Fixed.